plugins/hypervisors/xenserver/src/main/java/com/cloud/hypervisor/xenserver/resource/CitrixHelper.java

```diff
@@ -44,4 +46,12 @@ public static String getPVbootloaderArgs(String guestOS) {
         }
         return "";
     }
+
+    public static String getSRNameLabel(final StorageFilerTO primaryStore) {
+        if (Storage.StoragePoolType.PreSetup.equals(primaryStore.getType()) &&
+                !primaryStore.getPath().contains(primaryStore.getUuid())) {
+            return  primaryStore.getPath().replace("/", "");
```

Comment:
doesn't this replace all '/'? seems dangerous to me

Reply:
https://github.com/apache/cloudstack/blob/4.15/plugins/storage/volume/default/src/main/java/org/apache/cloudstack/storage/datastore/lifecycle/CloudStackPrimaryDataStoreLifeCycleImpl.java#L340-L342

Primary store setup code uses the same. I can change this though.
For pre-setup SR path is `/SR_NAME_LABEL` so it might not be as dangerous as it looks.